src/bin/main.ml

```diff
@@ -92,7 +92,43 @@ let set_signal_if_supported signal handler =
   with Invalid_argument _ ->
     ()
 
-let start urls sandbox git =
+module Date = struct
+  let pretty d =
+    let tm = Unix.localtime (Int64.to_float d) in
+    Printf.sprintf "%02d:%02d:%02d"
+      tm.Unix.tm_hour tm.Unix.tm_min tm.Unix.tm_sec
+end
+
+module HTTP = Irmin_http_server.Make(Cohttp_lwt_unix.Server)(Date)
+
+let serve_http = ()
+
+let http_server uri git =
+  let timeout = 3600 in
+  let uri = match Uri.host uri with
+    | None   -> Uri.with_host uri (Some "localhost")
+    | Some _ -> uri in
+  let port, uri = match Uri.port uri with
+    | None   -> 8080, Uri.with_port uri (Some 8080)
+    | Some p -> p, uri in
+  let mode = `TCP (`Port port) in
+  Logs.info (fun f -> f "daemon: %s" (Uri.to_string uri));
+  Printf.printf "Server starting on port %d.\n%!" port;
+  begin match git with
+    | None -> (* in-memory store *)
+      let module HTTP = HTTP(In_memory_store.Store) in
```

Comment:
Why are we creating the store inside the HTTP code? Shouldn't the store be shared?